test/performance/odp_packet_gen.c

```diff
@@ -1032,6 +1066,27 @@ static int close_pktios(test_global_t *global)
 	return ret;
 }
 
+static inline void get_timestamp(odp_packet_t pkt, uint32_t ts_off, rx_lat_data_t *lat_data)
+{
+	ts_data_t ts_data;
+	uint64_t nsec;
+
+	if (odp_unlikely(odp_packet_copy_to_mem(pkt, ts_off, sizeof(ts_data), &ts_data) < 0 ||
+			 ts_data.magic != TS_MAGIC))
+		return;
+
+	nsec = odp_time_global_ns() - ts_data.tx_ts;
```

Comment:
If a burst of packets is received, a more accurate method for time stamping would be to save the time once right after receiving the events and then using the same RX time for all packets in the burst. This would also have less overhead.

The same style could also be used in the TX side to reduce overhead, although it could strictly speaking be a bit  less accurate.

Reply:
Saving time once in RX side in V3. For better accuracy, no point then making it less accurate in TX side, overhead of the timestamping seems negligible.